packages/shared/src/authz/policies/project/tokens/canCreateEmbedToken.spec.ts

```diff
@@ -0,0 +1,133 @@
+import cryptoRandomString from 'crypto-random-string'
+import { Roles } from '../../../../core/constants.js'
+import { parseFeatureFlags } from '../../../../environment/index.js'
+import { getProjectFake, getWorkspaceFake } from '../../../../tests/fakes.js'
+import { canCreateEmbedTokenPolicy } from './canCreateEmbedToken.js'
+import { assert, describe, expect, it } from 'vitest'
+import {
+  ProjectNotEnoughPermissionsError,
+  ServerNoAccessError,
+  WorkspacePlanNoFeatureAccessError
+} from '../../../domain/authErrors.js'
+
+const buildCanCreateEmbedToken = (
+  overrides?: Partial<Parameters<typeof canCreateEmbedTokenPolicy>[0]>
```

Comment:
these should test both workspaced non-workspaced contexts

Reply:
imo the relevant workspace/non-workspace cases are handled in the overrides in the tests, can be tweaked if you don't feel so